File: Microservice Web/MicroserviceWeb/src/app/shared/invoice.ts
import {InvoiceRequest} from "./invoice-request";
import {Customer} from "./customer";

export class Invoice {

  id:any;
  invoiceDate:Date | undefined;
  totalSum:any | undefined;
  status: boolean | undefined;
  orders: InvoiceRequest[] | undefined
  customer: Customer | undefined
  customerId: any| undefined
}

File: Microservice Back/Invoice-service/src/main/java/org/ms/invoiceservice/model/Invoice.java
package org.ms.invoiceservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.util.Collection;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Invoice {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date invoiceDate;
    private Double totalSum;
    private boolean status;
    @OneToMany(mappedBy = "invoice")
    private Collection<OrderItem> Orders;
    @Transient
    private Customer customer;
    private Long customerId;
    private String customerName;

}

